spdmlib/src/crypto/spdm_ring/asym_verify_impl.rs

```diff
@@ -80,7 +80,9 @@ fn asym_verify(
             match base_asym_algo {
                 SpdmBaseAsymAlgo::TPM_ALG_ECDSA_ECC_NIST_P256
                 | SpdmBaseAsymAlgo::TPM_ALG_ECDSA_ECC_NIST_P384 => {
-                    let mut der_signature = [0u8; 66 * 2 + 8 + 1];
+                    // DER has this format: 0x30 size 0x02 r_size 0x00 [r_size] 0x02 s_size 0x00 [s_size]
+                    let mut der_signature =
+                        [0u8; crate::protocol::ECDSA_ECC_NIST_P384_KEY_SIZE + 8];
```

Comment:
why you change "+ 8 + 1" to "+ 8" ?

Reply:
> why you change "+ 8 + 1" to "+ 8" ?

what does the "+ 1" mean? the DER format should be something like `0x30 size 0x02 r_size 0x00 [r_size] 0x02 s_size 0x00 [s_size]`, the [r_size] + [s_size] should less than or equal to crate::protocol::ECDSA_ECC_NIST_P384_KEY_SIZE, the 8 is for `0x30 size 0x02 r_size 0x00 0x02 s_size 0x00`. Don't know why need to + 1 more byte.

Reply:
if the first byte starts from 0x80, then we need append 0x00 before it.

Reply:
> if the first byte starts from 0x80, then we need append 0x00 before it.

the two 0x00 in this `0x30 size 0x02 r_size 0x00 [r_size] 0x02 s_size 0x00 [s_size]` are optianal, and exist only when the first byte >= 0x80. So this case is already counted in for this purpose. We don't need the + 1.

Reply:
OK. It seems +8 is right.
I forget why it is +8+1 before.